src/ai_migrate/storage/config_loader.py

```diff
@@ -0,0 +1,178 @@
+import os
+import yaml
+from pathlib import Path
+from typing import Optional, Dict, Union
+
+from .config import StorageConfig
+
+DEFAULT_CONFIG_NAME = ".ai-migrate.yml"
+PROJECT_MARKERS = [
+    "pyproject.toml",
+    "setup.py",
+    "setup.cfg",
+    ".ai-migrate-root",
+    DEFAULT_CONFIG_NAME,
+]
+
+DEFAULT_CONFIG_CONTENT = {"type": "local", "path": "migrations"}
+
+
+def find_project_root(start_path: Union[str, Path] = None) -> Optional[Path]:
+    """Find the project root by looking for project marker files.
+
+    @param start_path Starting path for the search (default: current directory)
+    @return Path to project root or None if not found
+    """
+    current = Path(start_path or os.getcwd()).resolve()
+
+    while current != current.parent:
+        # Check for any of the marker files
+        for marker in PROJECT_MARKERS:
```

Comment:
Is this the right place to look? This seems like it would expect you to be in a python project, but users will run the tool from all types of projects. We do assume a git repo though.

Is the intention that the user overrides this per-project?

Reply:
yeah I am still working on this one.